src/main/java/core/basesyntax/service/impl/CsvFormatReportGenerator.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.service.ReportGenerator;
+import java.util.Map;
+
+public class CsvFormatReportGenerator implements ReportGenerator {
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+    private static final String CSV_SEPARATOR = ",";
+    private static final String START_MESSAGE = "fruit,quantity";
+    private final FruitStorageDao storageDao;
+
+    public CsvFormatReportGenerator(FruitStorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public String create() {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(START_MESSAGE);
+
+        storageDao.getAll().stream()
+                .map(Map.Entry::getKey)
+                .forEach(e -> stringBuilder.append(LINE_SEPARATOR)
+                        .append(e)
+                        .append(CSV_SEPARATOR)
+                        .append(storageDao.getQuantity(e)));
```

Comment:
In the `create` method, you are calling `storageDao.getQuantity(e)` inside the `forEach` loop. This can be inefficient if `getQuantity` involves complex operations or if the storage is large. Since you already have the entries from `getAll()`, consider using `Map.Entry::getValue` to directly access the quantity, which would be more efficient.